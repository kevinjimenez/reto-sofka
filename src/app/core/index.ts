export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
