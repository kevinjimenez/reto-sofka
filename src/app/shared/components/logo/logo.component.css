.logo {
  width: 2rem;
  height: 2rem;
  background-color: #cbd5e0;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}
