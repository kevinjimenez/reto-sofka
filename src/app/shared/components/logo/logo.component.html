@if (!hasError()) {
	<img class="logo" [src]="path()" alt="__logo__" onerror="" (error)="setError()" />
} @else {
	<div class="logo">LG</div>
}
