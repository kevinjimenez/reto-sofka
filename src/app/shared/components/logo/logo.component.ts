import { Component, Input, signal } from '@angular/core';
import { ImagePipe } from '../../pipes';

@Component({
	selector: 'app-logo',
	standalone: true,
	imports: [ImagePipe],
	templateUrl: './logo.component.html',
	styleUrl: './logo.component.css'
})
export class LogoComponent {
	public path = signal<string>('https://placehold.co/600x400');
	public default = signal<string>('photo.svg');
	public hasError = signal<boolean>(false);
	@Input() set image(value: string) {
		if (value) {
			this.path.set(value);
		}
	}

	public setError() {
		this.hasError.set(true);
	}
}
