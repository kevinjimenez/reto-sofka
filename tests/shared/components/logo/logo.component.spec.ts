import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LogoComponent } from '../../../../src/app/shared/components/logo/logo.component';

describe('LogoComponent', () => {
	let component: LogoComponent;
	let fixture: ComponentFixture<LogoComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [LogoComponent]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(LogoComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should set path to default value when image input is not set', () => {
		const contend = compiled.querySelector('img');
		expect(contend).toBeTruthy();
		expect(component.path()).toBe('https://placehold.co/600x400');
	});

	it('should set path', () => {
		const expeected = 'https://example.com/image.png';
		component.image = expeected;
		fixture.detectChanges();

		const contend = compiled.querySelector('img');
		expect(contend).toBeTruthy();
		expect(component.path()).toBe(expeected);
	});

	it('should render default content when load img error', () => {
		component.image = '://example.com/image.png';
		component.setError();
		fixture.detectChanges();

		const contend = compiled.querySelector('img');
		expect(contend).toBeTruthy();
	});
});
