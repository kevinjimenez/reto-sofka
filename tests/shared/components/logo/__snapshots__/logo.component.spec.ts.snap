// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <img
    alt="__logo__"
    class="logo"
    onerror=""
    src="https://placehold.co/600x400"
  />
  
  
</div>
`;
